optimize.py

```diff
@@ -1,51 +1,77 @@
 import sys
-import os
+import argparse
 import pennylane as qml
 from pennylane import numpy as np
 import torch
 import gc
 from timeit import default_timer as timer
 from datetime import datetime, timedelta
 
-from utils import clustered_chain_graph, get_qaoa_circuit
+from utils import clustered_chain_graph, get_qaoa_circuit, grad_reduction_qaoa_circuit
 
-from ray import train
+import ray
 
-########################################################################
-# Execution env setup
-########################################################################
-split = 1 # number of gpus per tape execution (can be fractional)
+seed = 1967
+split_gpu = 0  # number of gpus per tape execution (can be fractional). Set as zero for cpu only.
 
-r = 3 # number of clusters
-n = 20  # nodes in clusters
-k = 1  # vertex separators
 
-layers = 2 
+def parse_args():
+    parser = argparse.ArgumentParser()
 
-time_stamp = datetime.now().strftime('%Y-%m-%d-%H:%M:%S')
-filename = f"./data/optimisation/opt_p={layers}_r={r}_n={n}_k={k}_{time_stamp}"
-sys.stdout = open(filename, 'w')
+    parser.add_argument(
+        "--device_name",
+        type=str,
+        default="lightning.qubit",
+        const="lightning.qubit",
+        nargs="?",
+        choices=("lightning.qubit", "lightning.gpu", "lightning.kokkos", "default.qubit"),
```

Comment:
"lightning.gpu" and "lightning.kokkos" need to be tested.

Reply:
I will make the `split_gpu` a variable controllable when calling the script, as those.
The reason is that we can't be sure if `lightning.kokkos` are being called to run with CPU or GPU, also it will probably be cool to be able to play with this parameter more easily. The default will be zero.